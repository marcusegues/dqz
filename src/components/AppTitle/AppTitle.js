import React from 'react';
import { View, Text, Image } from 'react-native';
import { moderateScale, scale } from '../../styles/Scaling';

const img = require('../../../assets/images/logo_red.png');

const ownStyles = {
  titleWrapper: {
    flexDirection: 'row',
    marginLeft: 10,
  },
  appTitleContainer: {
    flexDirection: 'column',
    justifyContent: 'center',
  },
  appTitleWe: {
    color: '#1A1A1A',
    fontFamily: 'roboto_bold',
    fontSize: moderateScale(47),
    letterSpacing: 7,
  },
  appTitle: {
    fontFamily: 'exo_extraLight',
    color: '#1A1A1A',
    fontSize: moderateScale(32),
    letterSpacing: 8,
    fontWeight: '200',
  },
  logo: {
    width: null,
    height: scale(60),
    marginVertical: 10,
  },
};

const AppTitle = () => (
  <View style={ownStyles.appTitleContainer}>
    <Image source={img} style={ownStyles.logo} resizeMode="contain" />
    <View style={ownStyles.titleWrapper}>
      <Text style={ownStyles.appTitle}>CLARA</Text>
    </View>
  </View>
);

export default AppTitle;
