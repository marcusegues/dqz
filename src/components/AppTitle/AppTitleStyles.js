import { StyleSheet } from 'react-native';
import { moderateScale, scale } from '../../styles/Scaling';

export default StyleSheet.create({
  titleWrapper: {
    flexDirection: 'row',
    marginLeft: 10,
  },
  appTitle: {
    flexDirection: 'row',
    justifyContent: 'center',
  },
  appTitleWe: {
    color: '#1A1A1A',
    fontFamily: 'roboto_bold',
    fontSize: moderateScale(47),
  },
  appTitleDeclare: {
    fontFamily: 'roboto_light',
    fontSize: moderateScale(47),
  },
  logo: {
    width: scale(42),
    height: null,
  },
});
