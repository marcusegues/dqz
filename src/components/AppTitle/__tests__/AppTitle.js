import renderer from 'react-test-renderer';
import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme/build/index';
import { AppTitle } from '../AppTitle';

describe('AppTitle', () => {
  test('renders according to snapshot', () => {
    const component = renderer.create(<AppTitle />).toJSON();
    expect(component).toMatchSnapshot();
  });
  test('and it says Zoll-App', () => {
    const component = shallow(<AppTitle />);
    expect(component.containsMatchingElement(<Text>Zoll-App</Text>)).toBe(true);
  });
});
