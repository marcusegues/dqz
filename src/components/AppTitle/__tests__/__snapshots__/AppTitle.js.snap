// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppTitle renders according to snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Object {
        "height": 150,
        "marginBottom": 8,
        "marginTop": 17,
        "width": null,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginLeft": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1A1A1A",
          "fontFamily": "exo_extraLight",
          "fontSize": 50.285714285714285,
          "fontWeight": "200",
          "letterSpacing": 8,
        }
      }
    >
      Zoll-App
    </Text>
  </View>
</View>
`;
